tests/test_chatinvitelink.py

```diff
@@ -107,6 +107,33 @@ def test_de_json_all_args(self, bot, creator):
         assert invite_link.name == self.name
         assert invite_link.pending_join_request_count == self.pending_join_request_count
 
+    def test_de_json_localization(self, tz_bot, bot, raw_bot, creator):
+        json_dict = {
+            "invite_link": self.link,
+            "creator": creator.to_dict(),
+            "creates_join_request": self.creates_join_request,
+            "is_primary": self.primary,
+            "is_revoked": self.revoked,
+            "expire_date": to_timestamp(self.expire_date),
+            "member_limit": self.member_limit,
+            "name": self.name,
+            "pending_join_request_count": str(self.pending_join_request_count),
+        }
+
+        invite_link_raw = ChatInviteLink.de_json(json_dict, raw_bot)
+        invite_link_bot = ChatInviteLink.de_json(json_dict, bot)
+        invite_link_tz = ChatInviteLink.de_json(json_dict, tz_bot)
+
+        # comparing utcoffsets because comparing timezones is unpredicatable
+        invite_offset = datetime.datetime.utcoffset(invite_link_tz.expire_date)
```

Comment:
```suggestion
        invite_offset = invite_link_tz.expire_date.utcoffset()
```

`datetime.datetime.utfcoffset` is a bound method rather than a class method, so the argument that you pass to it is actually the `self`. That works, but feels strange to me … The SO link that I pointed you to also does that "wrongly".